### ISSUE TITLE
Rewrite Data Classes Using Pydantic

### ISSUE THREAD

**Issue by SimonBoothroyd** [shirtsgroup/physical_validation#71]
## Description

We may consider re-writing the different data classes (such as `SimulationData` and `SystemData`) using [pydantic](https://pydantic-docs.helpmanual.io/) to cut down on the amount of 'boiler-plate' code, make everything more readily (de)serialisable and to make everything a bit more standardised.

**Comment by mattwthompson** [comment]
Interested in this as it overlaps with OpenFF work but happy to defer to an interested grad student

**Comment by SimonBoothroyd** [comment]
Possibly of use: https://github.com/samuelcolvin/pydantic/issues/380 (see the last comment in particular).

**Comment by ptmerz** [comment]
This is a great suggestion, I wish this had been available when I started to work on this! Definitely in favor! I would argue, however, that we should work on having at least a minimal set of tests before doing large infrastructural changes like this one - just to have some confidence in approving PRs.

**Comment by mattwthompson** [comment]
Agreed; in my opinion, tests are quite literally the most important thing to do right now. This is an idea for little farther down the road

**Comment by SimonBoothroyd** [comment]
> I would argue, however, that we should work on having at least a minimal set of tests before doing large infrastructural changes like this one

I definitely agree!
